openedx/openedx-platform
# feat: Add REST APIs for course advanced settings and course tabs [BD-38] [TNL-8625] [BB-4132]

This PR adds new APIs that allow MFEs to access and update course advanced settings, and to get the course tabs and show/hide/reorder them. 

**Sandbox URL**: 

*  LMS: https://pr27608.sandbox.opencraft.hosting/
*  Studio: https://studio.pr27608.sandbox.opencraft.hosting/

**Merge deadline**: "None" 

**Testing instructions**:

1. Run the added unit tests. 
2. You can see the data about all available course advanced settings using the same format used by studio by visiting: http://localhost:18010/api/contentstore/v0/advanced_settings/course-v1:edX+DemoX+Demo_Course
3. You can make a post request to the above using data formatted as: 
```json
{
  "disable_progress_graph": {
     "value": false  
   }
}
```
4.  Check that the list of tabs and their settings in the course using the tabs api: http://localhost:18010/api/contentstore/v0/tabs/course-v1:edX+DemoX+Demo_Course
5. Try hiding a tab by posting ``{"is_hidden": true"}` to http://localhost:18010/api/contentstore/v0/tabs/course-v1:edX+DemoX+Demo_Course/settings?tab_id=progress
6. The tab should be hidden from the UI
7. Test rearranging tabs by posting a new tab order to http://localhost:18010/api/contentstore/v0/tabs/course-v1:edX+DemoX+Demo_Course/reorder eg:
```json
[
    {
        "tab_id": "info"
    },
    {
        "tab_id": "courseware"
    },
    {
        "tab_id": "wiki"
    },
    {
        "tab_id": "discussion"
    },
    {
        "tab_id": "progress"
    }
]
```

**Reviewers**

[TNL-8625](https://openedx.atlassian.net/browse/TNL-8625)
